lib/engine/game/base.rb

```diff
@@ -31,38 +31,44 @@
 
 module Engine
   module Game
-    def self.load(data, at_action: nil, **kwargs)
-      game_data =
-        case data
-        when String
-          return load(JSON.parse(File.exist?(data) ? File.read(data) : data), at_action: at_action, **kwargs)
-        when Hash
-          {
-            title: data['title'],
-            names: data['players'].map { |p| [p['id'] || p['name'], p['name']] }.to_h,
-            id: data['id'],
-            actions: data['actions'] || [],
-            pin: data.dig('settings', 'pin'),
-            optional_rules: data.dig('settings', 'optional_rules') || [],
-          }
-        when Integer
-          return load(::Game[data], at_action: at_action, **kwargs)
-        when ::Game
-          {
-            title: data.title,
-            names: data.ordered_players.map { |u| [u.id, u.name] }.to_h,
-            id: data.id,
-            actions: data.actions.map(&:to_h),
-            pin: data.settings['pin'],
-            optional_rules: data.settings['optional_rules'] || [],
-          }
-        end
-      title = game_data.delete(:title)
-      names = game_data.delete(:names)
-      game_data.merge!(kwargs)
```

Comment:
I ended up deciding I didn't like dealing with the merge and deleting certain items to avoid them merging with kwargs, so I got rid of the hash and just explicitly set all of the relevant variables. `strict` and `disable_user_errors` can still be passed as kwargs to `load` and they will be forwarded on to `initialize`